lightning/src/sign/ecdsa.rs

```diff
@@ -39,8 +48,12 @@ pub trait EcdsaChannelSigner: ChannelSigner {
 	///
 	/// Note that all the relevant preimages will be provided, but there may also be additional
 	/// irrelevant or duplicate preimages.
-	//
-	// TODO: Document the things someone using this interface should enforce before signing.
+	///
+	/// An `Err` can be returned to signal that the signer is unavailable/cannot produce a valid
+	/// signature and should be retried later. Once the signer is ready to provide a signature after
+	/// previously returning an `Err`, [`ChannelManager::signer_unblocked`] must be called.
```

Comment:
There's a sentence ~10 lines up "Note that if signing fails or is rejected, the channel will be force-closed." I think it's fairly clear as-is but we could consider folding that sentence into this paragraph instead to avoid sounding contradictory.